### PR TITLE
Fix flow of CompilerGeneratedFilesOutputPath from CPS

### PR DESCRIPTION
**Pull request by tmat**
Fixes https://github.com/dotnet/roslyn/issues/75680

### CODE REVIEW

**Review by ToddGrun — APPROVED** [review]

**Review by arunchndr — APPROVED** [review]

**Comment by arunchndr** [comment]
@akhera99 fyi

**Comment by drewnoakes** [comment]
Linking to #75311

File: src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs [hunk 1]
```diff
@@ -199,7 +199,7 @@ public void SetProperty(string name, string? value)
             }
             else if (name == BuildPropertyNames.CompilerGeneratedFilesOutputPath)
             {
-                _projectSystemProject.GeneratedFilesOutputDirectory = string.IsNullOrWhiteSpace(value) ? null : value;
+                _projectSystemProject.GeneratedFilesOutputDirectory = PathUtilities.IsAbsolute(value) ? value : null;
```
Comment (drewnoakes) [inline src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs:202]: If a user sets `EmitCompilerGeneratedFiles` to `true` in their project, then `CompilerGeneratedFilesOutputPath` evaluates to a relative path by default. With this change, the value will be ignored, which seems like an issue.

I don't see any docs for `CompilerGeneratedFilesOutputPath`, but all the examples I see online use a relative value:.

- https://github.com/dotnet/roslyn/issues/49981
- https://stackoverflow.com/a/68320185/24874
- https://www.reddit.com/r/dotnet/comments/mrgx3u/how_to_put_source_generator_code_into_source/

**Review by chsienki — APPROVED** [review]

File: src/VisualStudio/CSharp/Test/ProjectSystemShim/CPS/AdditionalPropertiesTests.cs [hunk 2]
```diff
@@ -161,13 +161,19 @@ public async Task SetProperty_CompilerGeneratedFilesOutputPath_CPS()
             {
                 Assert.Null(environment.Workspace.CurrentSolution.Projects.Single().CompilationOutputInfo.GeneratedFilesOutputDirectory);
 
-                // relative path ignored:
+                // relative path is realtive to the project dir:
```
Comment (chsienki) [inline src/VisualStudio/CSharp/Test/ProjectSystemShim/CPS/AdditionalPropertiesTests.cs:164]:
```suggestion
                // relative path is relative to the project dir:
```

File: src/VisualStudio/CSharp/Test/ProjectSystemShim/CPS/CSharpCompilerOptionsTests.cs [hunk 3]
```diff
@@ -200,6 +200,12 @@ public async Task CompilerGeneratedFilesOutputPath_CPS()
             cpsProject.SetOptions(["/generatedfilesout:" + path]);
 
             AssertEx.AreEqual(path, environment.Workspace.CurrentSolution.Projects.Single().CompilationOutputInfo.GeneratedFilesOutputDirectory);
+
+            // relative path is realtive to the project dir:
```
Comment (chsienki) [inline src/VisualStudio/CSharp/Test/ProjectSystemShim/CPS/CSharpCompilerOptionsTests.cs:204]:
```suggestion
            // relative path is relative to the project dir:
```

**Review by jasonmalinowski — APPROVED** [review]

**Pull request merged**